src/main/plugins/org.dita.html5/xsl/get-meta.xsl

```diff
@@ -244,24 +245,26 @@ See the accompanying LICENSE file for applicable license.
   </xsl:template>
 
   <!-- CONTENT: Subject - prolog/metadata/keywords -->
-  <xsl:template match="*" mode="gen-keywords-metadata">
-    <xsl:variable name="keywords-content">
-      <!-- for each item inside keywords (including nested index terms) -->
-      <xsl:for-each select="descendant::*[contains(@class,' topic/prolog ')]/*[contains(@class,' topic/metadata ')]/*[contains(@class,' topic/keywords ')]/descendant-or-self::* |
-                            descendant::*[contains(@class,' map/topicmeta ')]/*[contains(@class,' topic/keywords ')]/descendant-or-self::* |
-                            descendant::*[contains(@class,' map/topicmeta ')]/*[contains(@class,' topic/metadata ')]/*[contains(@class,' topic/keywords ')]/descendant-or-self::*">
-        <!-- If this is the first term or keyword with this value -->
-        <xsl:if test="generate-id(key('meta-keywords',text()[1])[1]) = generate-id(.)">
-          <xsl:if test="position() > 2">
-            <xsl:text>, </xsl:text>
-          </xsl:if>
-          <xsl:value-of select="normalize-space(text()[1])"/>
-        </xsl:if>
-      </xsl:for-each>
-    </xsl:variable>
+  <xsl:template match="*[contains(@class, ' topic/topic ')]" mode="gen-keywords-metadata">
+    <xsl:variable name="keywords" as="element()*"
+                  select="*[contains(@class,' topic/prolog ')]/
```

Comment:
I don't think we should turn off the descendant matching for prolog here. Some of the background covered in this comment: https://github.com/dita-ot/dita-ot/pull/4410#issuecomment-1962357794

But basically - this code has been around for a long time, and -- the memories are hazy -- but I'm pretty confident this was an explicit reaction to user complaints when we did *not* get descendants. Topics are designed for reuse, and when you pull 3 topics into one file for HTML output, you only have one spot to put the keywords. You've created an HTML document where your content explicitly defines keywords for all three, say `keyword-topic1`, `keyword-topic2`, and `keyword-topic3`. Those should all apply because the meta tag applies to the whole document. If we only grab from the root topic, then we're explicitly dropping the keywords from those nested topics; search engines that still pick up this meta tag (if any still do that) will be losing important information about what's in the document.

Unlike in maps ... this isn't something that can really be "fixed" in the source. Most likely, the keywords from nested topics do *not* apply to the body of the root topic, so it's improper to move them up to that level. If the topics were combined through chunking, there is even less of an option.

Reply:
Good catch, deleted that by accident.